### PR TITLE
Parallel build

### PR DESCRIPTION
**Pull request by zbeekman**
Work around to allow for parallel compilation.

TL;DR: Put all interfaces and global modules into an object library. Then use generator expression to include the object files in the library along with the implementation sources.

In general, a better idea would be to create two libraries:

1. Interface library. Modifications to sources trigger recompilations
2. Implementation library. Modifications to sources do NOT trigger recompilations

Then create `PRIVATE` linkage from the implementation library to the interface library: This ensures that the build order is correct. Then, specify an `INTERFACE` linkage from the interface library to the implementation library. This *should* cause the build order to be correct, and then consumers only need to link against the interface library. This is untested as of now, because, in this simple example, the only target we want to build is a single static lib.

__N.B.__: The approach described in the last paragraph is not what I have implemented in this PR and is, as of now, untested.

### CODE REVIEW

**Comment by zbeekman** [comment]
After this is pulled, we should pull porteri/vtkmofo:master into nrc-fuels/vtkmofo:master, then update submodules and fix FAST for parallel builds on transient solution.

**Review by porteri — APPROVED** [review]
Looks good to me
